Riot/Modules/Common/ActivityIndicator/GlobalActivityCenterPresenter.swift

```diff
@@ -0,0 +1,69 @@
+// 
+// Copyright 2021 New Vector Ltd
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+
+import Foundation
+import UIKit
+import MatrixSDK
+
+/// Activity indicator presenter which uses a shared `ActivityCenter` to coordinate different activity indicators,
+/// and which uses the root navigation view controller to display the activities.
+@objc final class GlobalActivityCenterPresenter: NSObject, ActivityIndicatorPresenterType {
+    private var loadingActivity: Activity?
+    
+    private var rootNavigationController: UINavigationController? {
```

Comment:
There has to be a better way to do this, but a lot of current ways rely on `LegacyAppDelegate`. Open to suggestions given existing coordinator / navigation architecture